packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -53,21 +55,100 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
         _requestItem: ReadAttributeRequestItem,
         params: AcceptReadAttributeRequestItemParametersJSON,
         requestInfo: LocalRequestInfo
-    ): Promise<ReadAttributeAcceptResponseItem> {
+    ): Promise<ReadAttributeAcceptResponseItem | AttributeSuccessionAcceptResponseItem> {
         const parsedParams = AcceptReadAttributeRequestItemParameters.from(params);
 
         let sharedLocalAttribute: LocalAttribute;
         if (parsedParams.isWithExistingAttribute()) {
-            sharedLocalAttribute = await this.copyExistingAttribute(parsedParams.existingAttributeId, requestInfo);
+            const existingSourceAttribute = await this.consumptionController.attributes.getLocalAttribute(parsedParams.existingAttributeId);
+            if (typeof existingSourceAttribute === "undefined") {
+                throw TransportCoreErrors.general.recordNotFound(LocalAttribute, parsedParams.existingAttributeId.toString());
+            }
+
+            const latestSharedVersion = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(
+                parsedParams.existingAttributeId,
+                [requestInfo.peer],
+                true
+            );
+
+            if (latestSharedVersion.length === 0) {
+                sharedLocalAttribute = await this.copyExistingAttribute(existingSourceAttribute.id, requestInfo);
+                return ReadAttributeAcceptResponseItem.from({
+                    result: ResponseItemResult.Accepted,
+                    attributeId: sharedLocalAttribute.id,
+                    attribute: sharedLocalAttribute.content
+                });
+            }
+
+            const predecessorOwnSharedAttribute = latestSharedVersion[0];
+            if (typeof predecessorOwnSharedAttribute.shareInfo?.sourceAttribute === "undefined") {
+                throw new Error(`The Attribute ${predecessorOwnSharedAttribute.id} does not fulfill the requirements of an own shared Attribute.`);
+            }
```

Comment:
can we also validate this, so that we don't
<img src="https://media4.giphy.com/media/KnF6GPtMbe2Fa/giphy.gif"/>
when accepting?

Reply:
Would you throw an error or return a failed validation result? This should never happen, since getSharedVersionsOfRepositoryAttribute queries for Attributes with a specific shareInfo.sourceAttribute, so it should never be undefined, but the compiler doesn't seem to know.

Reply:
ah, ok, then this is good as is! :)